src/main/java/io/numaproj/numaflow/function/FunctionServer.java

```diff
@@ -151,7 +155,23 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
                                 headers.get(FunctionConstants.DATUM_METADATA_WIN_START),
                                 FunctionConstants.WINDOW_END_TIME,
                                 headers.get(FunctionConstants.DATUM_METADATA_WIN_END));
-                return Contexts.interceptCall(context, call, headers, next);
+                ServerCall.Listener<ReqT> listener = Contexts.interceptCall(context, call, headers, next);
+                return new ForwardingServerCallListener.SimpleForwardingServerCallListener<>(listener) {
+                    @Override
+                    public void onHalfClose() {
+                        try {
+                            super.onHalfClose();
+                        } catch (RuntimeException ex) {
+                            handleException(ex, call, headers);
+                            throw ex;
+                        }
+                    }
+                    private void handleException(RuntimeException exception, ServerCall<ReqT, RespT> serverCall, Metadata headers) {
+                        // Currently, we only have application level exceptions.
+                        // Translate it to UNKNOWN status.
+                        serverCall.close(Status.UNKNOWN, headers);
+                    }
```

Comment:
what headers are we sending back?

Reply:
In the interceptor we are not adding anything to the headers. the original headers are sending back.

Reply:
Why do we need to send them back?

Reply:
LOL, this is part of required parameters for `serverCall.close()`

Reply:
Sorry, let me rephrase my question. Why are we sending the input headers back? Is it used on the client side? 
I know it's a required parameter, we can just go with null right?

Reply:
Got it, I don't see any reason we would like to update the original headers or send a null instead as you can see we are even adding datum metadata information to the headers at L155-157. Do you have any particular reason in mind?

Reply:
No reason, just wanted to know how it was used.